todoList/todoList/Services/TodoListService.cs

```diff
@@ -0,0 +1,117 @@
+﻿#nullable disable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using todoAPI.Data;
+using todoAPI.Model;
+
+namespace todoAPI.Services
+{
+    public class TodoListService : ITodoService
+    {
+        private readonly todoListContext _context;
+
+        public TodoListService(todoListContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<TodoList>> GetTodoList()
+        {
+            return await _context.TodoList.ToListAsync();
+        }
+
+        public async Task<TodoList> GetTodoItem(int id)
+        {
+            if (!TodoItemExists(id))
+            {
+                return null;
+            }
+            //var todoItem = await _context.TodoList.FindAsync(id);
+            return await _context.TodoList.Where(x => x.Id == id).Include(x => x.Tasks).FirstAsync();
```

Comment:
use FirstOrDefaultAsync instead FirstAsync because when not found FirstAsync will throw the error but FirstOrDefault will return null